### PR TITLE
[Mellanox] Add support for SN4600 system

### PR DESCRIPTION
**Pull request by DavidZagury**
#### Why I did it
Support new Mellanox system SN4600

#### How I did it
Add relevant files to support new platform names platform x86_64-mlnx_msn4600 and default SKU ACS-MSN4600

#### How to verify it
Load the SN4600 switch, verify all ports are up with 200G by default.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012

### CODE REVIEW

**Comment by liat-grozovik** [comment]
Suggest to change the PR title to something shorted such as "[Mellanox] Add support for SN4600 system"

**Comment by liat-grozovik** [comment]
Suggest to rephrase the comment to something like 
"
- Why I did it
Support new Mellanox system SN4600 

- How I did it
Add relevant files to support new platform names platform x86_64-mlnx_msn4600 and default SKU ACS-MSN4600 

- How to verify it
Load the SN4600 switch, verify all ports are up with 200G by default.
"

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py:47]: why adding it in the middle and not in as the last one?

Comment (liat-grozovik) [reply device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py:47]: why all has -r0 and only SN4600 and SN4600C does not?

File: device/mellanox/x86_64-mlnx_msn4600-r0/ACS-MSN4600/hwsku.json
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn4600-r0/ACS-MSN4600/hwsku.json:196]: This file is for the dynamic break out. meaning that we need to remove it from the basic PR so it can be merged to 202012 and have the DPB support on a seperated PR.

File: device/mellanox/x86_64-mlnx_msn4600-r0/ACS-MSN4600/buffers_dynamic.json.j2
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn4600-r0/ACS-MSN4600/buffers_dynamic.json.j2:1]: dont we have one for 4700? all are using 2700 only?

File: device/mellanox/x86_64-mlnx_msn4600-r0/ACS-MSN4600/port_config.ini
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn4600-r0/ACS-MSN4600/port_config.ini:1]: Are you sure this is aligned with hw? you need 4 lanes for 200G, Please check how it was done in Onyx.

File: device/mellanox/x86_64-mlnx_msn4600-r0/ACS-MSN4600/sai_4600.xml
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn4600-r0/ACS-MSN4600/sai_4600.xml:1]: please send the file to Itai to confirm it. Not sure if he has one to use on SAI github but at least  he need to confirm it

File: device/mellanox/x86_64-mlnx_msn4600-r0/platform_components.json
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn4600-r0/platform_components.json:14]: is this aligned with the HW mgmt definition? we have 3 CPLDs for this system?

File: device/mellanox/x86_64-mlnx_msn4600-r0/sensors.conf
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn4600-r0/sensors.conf:1]: please change to 2021

File: device/mellanox/x86_64-mlnx_msn4600-r0/sensors.conf
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn4600-r0/sensors.conf:189]: need to recheck this file once platform tests and sensor test is passing. until then we cannot approve it.
BTW i believe you should have asked Vadim P to review it ir the HW guy who worked with Shlomi on it for SN4600C.

File: platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py
Comment (liat-grozovik) [inline platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py:51]: why adding in the middle and not on the last?

File: platform/mellanox/mlnx-platform-api/sonic_platform/device_data.py
Comment (liat-grozovik) [inline platform/mellanox/mlnx-platform-api/sonic_platform/device_data.py:257]: for these numbers please review with Vadim P
we can only approve them after platform test is fully running on this system.

File: platform/mellanox/mlnx-platform-api/sonic_platform/thermal.py
Comment (liat-grozovik) [inline platform/mellanox/mlnx-platform-api/sonic_platform/thermal.py:137]: what is do you need a leading number in the comment?

**Review by liat-grozovik — CHANGES_REQUESTED** [review]

File: device/mellanox/x86_64-mlnx_msn4600-r0/ACS-MSN4600/sai_4600.xml
Comment (DavidZagury) [reply device/mellanox/x86_64-mlnx_msn4600-r0/ACS-MSN4600/sai_4600.xml:1]: We talked with Itay which told us that the file is already in github which I found here:
https://github.com/Mellanox/SAI-Implementation/blob/80937117fcd54e8df0581368f652f3664af6aecb/mlnx_sai/src/sai_4600.xml

But he did say it was a draft file.

File: device/mellanox/x86_64-mlnx_msn4600-r0/ACS-MSN4600/buffers_dynamic.json.j2
Comment (DavidZagury) [reply device/mellanox/x86_64-mlnx_msn4600-r0/ACS-MSN4600/buffers_dynamic.json.j2:1]: All the SPC3 switched uses the 2700, including the 4700

File: platform/mellanox/mlnx-platform-api/sonic_platform/thermal.py
Comment (DavidZagury) [reply platform/mellanox/mlnx-platform-api/sonic_platform/thermal.py:137]: The platform_dict_thermal is supposed to have the index of the switch thermal_profile_list, I added those to help me find the index of my switch in the array.

I kept it because it having this information in front of our eyes helped me realize there is a mistake in other switches (x86_64-mlnx_msn3420-r0 having 9 as its index when in fact it is in the 11th place of the array, 4410 which is in the 11 place of the list is missing from this array at all).

I can remove it if it is not needed

File: platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py
Comment (DavidZagury) [reply platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py:51]: The list was already ordered by the switch number and I wanted to keep this order

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py
Comment (DavidZagury) [reply device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py]: Fix for 4600, and talked with @shlomibitton and he will fix 4600C

File: device/mellanox/x86_64-mlnx_msn4600-r0/sensors.conf
Comment (DavidZagury) [reply device/mellanox/x86_64-mlnx_msn4600-r0/sensors.conf:189]: I talked with Vadim, he didn't have time for doing a full review but told me to check that it the same as the Leopard, I checked and they are only different on the fans, which he said that it is correct.

File: platform/mellanox/mlnx-platform-api/sonic_platform/device_data.py
Comment (DavidZagury) [reply platform/mellanox/mlnx-platform-api/sonic_platform/device_data.py:257]: I talked with Vadim, he didn't have time for doing a full review but told me to check that it the same as the Leopard, I checked and they are only different on the fans and unk, for those change he said that fans are correct, and unk should be the same as Tigon which it is.

File: device/mellanox/x86_64-mlnx_msn4600-r0/ACS-MSN4600/port_config.ini
Comment (DavidZagury) [reply device/mellanox/x86_64-mlnx_msn4600-r0/ACS-MSN4600/port_config.ini:1]: I check with Onyx and with HW and they confirmed that Liger uses 4 lanes for 200G

File: device/mellanox/x86_64-mlnx_msn4600-r0/pcie.yaml
Comment (shlomibitton) [inline device/mellanox/x86_64-mlnx_msn4600-r0/pcie.yaml:1]: please verify with kebo if this file is OK

File: device/mellanox/x86_64-mlnx_msn4600-r0/sensors.conf
Comment (shlomibitton) [reply device/mellanox/x86_64-mlnx_msn4600-r0/sensors.conf:189]: you can manually verify it with running a sensors dump command and then the sensors normal command.
check if the raw data we get and the normal data we get are aligned, I mean you don't have an output with the raw names for some of the components. All should be mapped with a user friendly names for human readable values.
raw output: "sensors -c /dev/null"
mapped output: "sensors"

File: platform/mellanox/mlnx-platform-api/sonic_platform/device_data.py
Comment (shlomibitton) [inline platform/mellanox/mlnx-platform-api/sonic_platform/device_data.py:275]: please add extra empty line at the end of the file

**Review by shlomibitton — COMMENTED** [review]
As commented

File: device/mellanox/x86_64-mlnx_msn4600-r0/platform_components.json
Comment (DavidZagury) [reply device/mellanox/x86_64-mlnx_msn4600-r0/platform_components.json:14]: According to the HW SPEC there are three CPLD devices in the system

File: device/mellanox/x86_64-mlnx_msn4600-r0/pcie.yaml
Comment (DavidZagury) [reply device/mellanox/x86_64-mlnx_msn4600-r0/pcie.yaml:1]: Talked with Kebo, the file is OK

**Review by shlomibitton — APPROVED** [review]

**Pull request closed without merging**
